### PR TITLE
[feat] blogs implemented

### LINKED ISSUE #3: blogs page
complete by 13 th Aug

### PR DESCRIPTION
**Pull request by riyaaaa123**
fix #3

### CODE REVIEW

File: backend/blog/models.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+# Create your models here.
+class Blog(JsonModel,TimestampedModel):
+    title=models.CharField(max_length=200)
+    content= RichTextField()
+    writer = models.ForeignKey(User,on_delete=models.CASCADE,related_name='blogs')
+    name= models.CharField(max_length=30)
+    date_posted= models.DateTimeField(default=timezone.now)
+    image_1= models.ImageField(upload_to=get_blog_pics_path())
+    Web_image= models.ImageField(upload_to=get_blog_pics_path())
```
Comment (pranav-leo) [inline backend/blog/models.py]: inconsit namming conv

File: backend/blog/views.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+from django.http import JsonResponse
+from django.contrib.postgres.search import TrigramSimilarity
+from rest_framework.exceptions import ParseError
+from utils.pagination import DefaultPagination
+from rest_framework.generics import ListCreateAPIView
+from rest_framework import status
+from rest_framework.response import Response
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.response import Response
+from rest_framework.status import HTTP_200_OK, HTTP_204_NO_CONTENT
```
Comment (pranav-leo) [inline backend/blog/views.py]: remove unused libraries

File: frontend/hooks/blogs.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  successCallback,
+  errorCallback,
+) => {
+  return useMutation(() => BlogData(page, pageSize, searchText), {
+    mutationKey: "get-blog-data",
+    onSuccess: (res) => {
+      successCallback(res);
+    },
+    onError: (err) => {
+      errorCallback(err, "hi");
```
Comment (pranav-leo) [inline frontend/hooks/blogs.js]: whom are sending hi to?

File: frontend/hooks/blogs.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    },
+  });
+};
+
+const SingleBlog = (id) => {
+  if (!id) return Promise.reject("ID is not provided");
```
Comment (pranav-leo) [inline frontend/hooks/blogs.js]: can remove this

File: frontend/package.json [hunk 5]
```diff
@@ -3,11 +3,15 @@
...
+    "dev": "next dev ",
     "build": "next build",
     "start": "next start",
-    "lint": "prettier --check ",
-    "format": "prettier --write "
+    "check-types": "tsc --pretty --noEmit",
+    "check-format": "prettier --check .",
+    "check-lint": "eslint . --ext ts --ext tsx --ext js",
+    "format": "prettier --write .",
+    "test-all": "npm run check-format && npm run check-lint && npm run check-types && npm run build",
```
Comment (pranav-leo) [inline frontend/package.json]: use yarn

File: frontend/pages/blog/index.tsx [hunk 6]
```diff
@@ -0,0 +1,186 @@
...
+      mutate();
+    }
+  }, [currentPage, searchText, mutate]);
+  const styleObj = {
+    background: "rgba(0, 0, 0, 0.10)",
+    borderRadius: "1.5rem",
+    boxShadow:
+      "2px 2px 2px 0px rgba(255, 255, 255, 0.05) inset, -1px -1px 4px 0px rgba(255, 255, 255, 0.05) inset, 0px 4px 4px 0px rgba(0, 0, 0, 0.10);",
+  };
+  const blogStyle = {
```
Comment (pranav-leo) [inline frontend/pages/blog/index.tsx:47]: remove unused

File: frontend/pages/blog/index.tsx [hunk 7]
```diff
@@ -0,0 +1,186 @@
...
+            <div className="flex px-3 py-3 w-[99%] " style={Search}>
+              <img src="./ri_search-line.svg" />
+              <input
+                type="text "
+                value={searchText}
+                onChange={(e) => {
+                  handleSearchInput(e);
+                }}
+                className=" w-[90%]  ml-3 text-2xl text-white-20  placeholder-white "
+                style={{ background: "rgba(84, 84, 84,1)", color: "white" }}
```
Comment (pranav-iitr) [inline frontend/pages/blog/index.tsx:89]: use tailwind

**Review by pranav-leo — CHANGES_REQUESTED** [review]

**Review by pranav-leo — APPROVED** [review]

**Pull request merged**